### PR TITLE
Allow Plural for Types in TagFilters

### PR DESCRIPTION
**Pull request by dave2wave**
closes: #2030 
closes: #1183

### CODE REVIEW

File: nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/tagging/TagFilter.java [hunk 1]
```diff
@@ -156,6 +156,9 @@ public TagFilter(String filterSpec) {
             for (String assignment : keyvalues) {
                 String[] keyvalue = assignment.split("[:=]", 2);
                 String key = keyvalue[0];
+                if (key.endsWith("s")) {
```
Comment (msmygit) [inline nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/tagging/TagFilter.java:159]: omg, that's a clever idea 💡 !!

File: nb-apis/nb-api/src/test/java/io/nosqlbench/nb/api/tagging/TagFilterTest.java [hunk 2]
```diff
@@ -77,6 +77,21 @@ public void testMatchingTagKeyValueDoesMatch() {
         assertThat(tf.matches(itemtags2).matched()).isTrue();
     }
 
+    @Test
```
Comment (msmygit) [inline nb-apis/nb-api/src/test/java/io/nosqlbench/nb/api/tagging/TagFilterTest.java:80]: love to see unit test cases!

**Review by msmygit — DISMISSED** [review]
LGTM 👍🏼

File: nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/tagging/TagFilter.java (hunk 1 above)
Comment (msmygit) [reply nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/tagging/TagFilter.java:159]: hopefully we don't have any legitimate keys that ends with `s`!

Comment (dave2wave) [reply nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/tagging/TagFilter.java:159]: We don't - there seem to be three categories defined in `OpDef.java`
```
        tagsWithName.put("block",block.getName());
        tagsWithName.put("name",this.rawOpDef.getName());
        tagsWithName.put("op",this.rawOpDef.getName());
```

**Comment by dave2wave** [comment]
Present detailed errors about matches.

**Comment by dave2wave** [comment]
Add NBAdvisor about the change.

**Review by jshook — APPROVED** [review]
LGTM!

**Pull request merged**